01-basics/20-weather/WeatherApp.js

```diff
@@ -4,44 +4,80 @@ import { getWeatherData, WeatherConditionIcons } from './weather.service.ts'
 export default defineComponent({
   name: 'WeatherApp',
 
+  setup() {
+    const data = getWeatherData()
+
+    function tempConverter(temp) {
+      const convertedTemp = (temp - 273.15).toFixed(1)
+      return convertedTemp
+    }
+
+    function pressureConverter(pressure) {
+      const convertedPressure = Math.round(pressure * 0.75)
+      return convertedPressure
+    }
+
+    function pickIcon(id) {
+      const icon = WeatherConditionIcons[id]
+      return icon
+    }
+
+    function isNightTheme({ dt, sunrise, sunset }) {
+      let currentTime = new Date()
+      currentTime.setHours(dt.split(':')[0], dt.split(':')[1])
+      let sunriseTime = new Date()
+      sunriseTime.setHours(sunrise.split(':')[0], sunrise.split(':')[1])
+      let sunsetTime = new Date()
+      sunsetTime.setHours(sunset.split(':')[0], sunset.split(':')[1])
+      return currentTime < sunriseTime || sunsetTime < currentTime
+    }
+
+    return {
+      data,
+      tempConverter,
+      pressureConverter,
+      pickIcon,
+      isNightTheme,
+    }
+  },
+
   template: `
     <div>
       <h1 class="title">Погода в Средиземье</h1>
-
       <ul class="weather-list unstyled-list">
-        <li class="weather-card weather-card--night">
-          <div class="weather-alert">
+        <li v-for="item in data" class="weather-card" :class="{'weather-card--night': isNightTheme({dt: item.current.dt, sunrise: item.current.sunrise, sunset:item.current.sunset })}">
+          <div v-if="item.alert" class="weather-alert">
             <span class="weather-alert__icon">⚠️</span>
-            <span class="weather-alert__description">Королевская метеослужба короля Арагорна II: Предвещается наступление сильного шторма.</span>
+            <span class="weather-alert__description">{{item.alert.sender_name}}: {{item.alert.description}}</span>
           </div>
           <div>
             <h2 class="weather-card__name">
-              Гондор
+              {{item.geographic_name}}
             </h2>
```

Comment:
It's preferred to always add spaces to a text interpolation. `{{ item.geographic_name }}`